src/api.h

```diff
@@ -296,6 +296,7 @@ typedef enum aws_crt_signed_body_header_type {
     AWS_CRT_SIGNED_BODY_HEADER_TYPE_NONE,
     AWS_CRT_SIGNED_BODY_HEADER_TYPE_X_AMZ_CONTENT_SHA256,
 } aws_crt_signed_body_header_type;
+typedef bool(aws_crt_should_sign_header_fn)(const struct aws_byte_cursor *name, void *userdata);
```

Comment:
we don't use aws_byte_cursors in our FFI API, this one could be a `const char* name` and `uint32_t length`

Reply:
The actual callback function defined in [signing_config.h](https://github.com/awslabs/aws-c-auth/blob/main/include/aws/auth/signing_config.h#L16) is an `aws_byte_cursor`.  Do we need to create some sort of adapter between them? (ie - a callback function that wraps the user provided and does the needed conversion?)

Reply:
I took care of that in my PR. you can abandon this one :)